new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java

```diff
@@ -5,86 +5,175 @@
+
+    // assertions for required parameters
+    assertThat(namespace)
+        .as("make sure namespace is not empty or null")
+        .isNotNull()
+        .isNotEmpty();
+
+    assertThat(params.getReleaseName())
+        .as("make sure releaseName is not empty or null")
+        .isNotNull()
+        .isNotEmpty();
+
+    assertThat(params.getRepoUrl() != null && params.getChartName() == null)
+        .as("make sure chart name is not empty or null when repo url is provided")
+        .isFalse();
+
+    assertThat(params.getRepoUrl() == null && params.getChartDir() == null)
+        .as("make sure repo url, chart name and chart dir are not empty or null. "
+            + "repo url, chart name or chart dir must be provided")
+        .isFalse();
+
+    //chart reference to be used in helm install
+    String chartRef = params.getChartDir();
+
+    // use repo url as chart reference if provided
+    if (params.getChartName() != null && params.getRepoUrl() != null) {
+      Helm.addRepo(params.getChartName(), params.getRepoUrl());
+      chartRef = params.getRepoUrl();
+    }
+
+    logger.info(String.format("Installing application in namespace %s using chart %s",
+        namespace, chartRef));
+
+    // build helm install command
+    StringBuffer installCmd = new StringBuffer("helm install ")
+        .append(params.getReleaseName()).append(" ").append(chartRef)
+        .append(" --namespace ").append(params.getNamespace());
 
-  public Helm values(HashMap<String, Object> values) {
-    this.values = values;
-    return this;
-  }
+    // add all the parameters
+    appendValues(params.getValues(), installCmd);
 
-  public String getChartName() {
-    return chartName;
-  }
+    installCmd.append(" && helm list");
+    // run the command
+    return exec(installCmd.toString());
 
-  public String getReleaseName() {
-    return releaseName;
   }
 
-  public String getNamespace() {
-    return namespace;
+  /**
+   * upgrade a helm release
+   * @param params the parameters to helm as values
+   * @return true on success, false otherwise
+   */
+  public static boolean upgrade(HelmParams params) {
+    String namespace = params.getNamespace();
+
+    // assertions for required parameters
+    assertThat(namespace)
+        .as("make sure namespace is not empty or null")
+        .isNotNull()
+        .isNotEmpty();
+
+    assertThat(params.getReleaseName())
+        .as("make sure releaseName is not empty or null")
+        .isNotNull()
+        .isNotEmpty();
+
+    assertThat(params.getChartDir())
+        .as("make sure chart dir is not empty or null")
+        .isNotNull()
+        .isNotEmpty();
+
+    //chart reference to be used in helm upgrade
+    String chartDir = params.getChartDir();
+
+    logger.info(String.format("Upgrade application in namespace %s using chart %s",
+        namespace, chartDir));
+
+    // build helm upgrade command
+    StringBuffer upgradeCmd = new StringBuffer("helm upgrade ")
```

Comment:
Its not a big deal, but this seems harder to read than necessary.  Why not use String.format?
String upgradeCmd = "helm upgrade {0} {1} --namespace {3}";

appendValues( params.getValues(),
   String.format(upgradeCmd, params.getReleaseName(), chartDir, params.getNamespace());

Reply:
yeah i think `String.format()` would be a lot cleaner to read

Reply:
will modify to use String.format()

Reply:
done